app/windows/PublishToIPNS/renderer.jsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react'
+import ReactDom from 'react-dom'
+
+import { remote } from 'electron'
+import { initIPFSClient, publishToIPNS, getPeer } from '../../api'
+import { trackEvent } from '../../stats'
+
+// Load Components
+import {
+  Window,
+  Content,
+  Input,
+  Toolbar,
+  Actionbar,
+  Button
+} from 'react-photonkit'
+import ProgressOverlay from '../../components/ProgressOverlay'
+import CircularProgress from '../../components/CircularProgress'
+
+class PublishToIPNSWindow extends React.Component {
+  state = {
+    loading: false
+  }
+
+  componentDidMount () {
+    trackEvent('PublishToIPNSWindowOpen')
+    getPeer().then(peer => this.setState({ peer }))
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault()
+    this.setState({ loading: true })
+    const hash = event.target[1].value
+    publishToIPNS(hash)
+      .then(result => {
+        this.setState({ loading: false })
+        const message = `Successfully published ${result.value} to IPNS!`
```

Comment:
Should we add the IPNS hash instead?
Something like `IPNS $HASH has been successfully updated to ${result.value}`?